src/components/ec-modal/ec-modal.vue

```diff
@@ -99,132 +99,123 @@
-      return !!this.isLoading.positive;
-    },
-    isLoadingNegativeButton() {
-      return !!this.isLoading.negative;
-    },
-    positiveButtonCategory() {
-      return this.category.positive || 'primary';
-    },
-    negativeButtonCategory() {
-      return this.category.negative || 'secondary';
-    },
-    zIndexStyle() {
-      return this.zIndex ? { zIndex: this.zIndex } : null;
-    },
+  large: {
+    type: Boolean,
+    default: false,
   },
-  watch: {
-    show: {
-      immediate: true,
-      handler(value) {
-        if (value) {
-          document.addEventListener('keyup', this.escapeIsPressed);
-        } else {
-          document.removeEventListener('keyup', this.escapeIsPressed);
-        }
-      },
+  isLoading: {
+    type: Object,
+    default: () => ({}),
+  },
+  zIndex: {
+    type: Number,
+    validator(value) {
+      return value > 200 && value < 250;
     },
   },
-  beforeUnmount() {
-    document.removeEventListener('keyup', this.escapeIsPressed);
+  category: {
+    type: Object,
+    default: () => ({}),
   },
-  methods: {
-    negativeAction() {
-      this.$emit('negative');
-    },
-    positiveAction() {
-      this.$emit('positive');
-    },
-    closeModal() {
-      if (this.isClosable) {
-        this.$emit('update:show', false);
-        this.$emit('close', false);
-      }
-    },
-    escapeIsPressed(e) {
-      if (e.keyCode === KeyCode.ESC) {
-        this.closeModal();
-      }
-    },
-    getFocusTrapOptions() {
-      const options = {
-        escapeDeactivates: this.isClosable,
-        clickOutsideDeactivates: this.isClosable,
-      };
-
-      if (this.hasPositiveButton()) {
-        options.initialFocus = /* istanbul ignore next */ () => this.$refs.positiveButton.$el;
-      } else if (this.hasNegativeButton()) {
-        options.initialFocus = /* istanbul ignore next */ () => this.$refs.negativeButton.$el;
-      }
+});
 
-      return options;
-    },
-    hasPositiveButton() {
-      return !!this.$slots.positive;
-    },
-    hasNegativeButton() {
-      return !!this.$slots.negative;
-    },
-    hasFooterLeftContent() {
-      return !!this.$slots.footerLeftContent;
-    },
-    hasFooter() {
-      return this.hasFooterLeftContent() || this.hasPositiveButton() || this.hasNegativeButton();
-    },
-  },
-};
+const emit = defineEmits(['update:show', 'negative', 'positive', 'close']);
+
+const positiveButton = ref(null);
+const negativeButton = ref(null);
+
+const isLoadingPositiveButton = computed(() => !!props.isLoading.positive);
+const isLoadingNegativeButton = computed(() => !!props.isLoading.negative);
+const positiveButtonCategory = computed(() => props.category.positive || 'primary');
+const negativeButtonCategory = computed(() => props.category.negative || 'secondary');
+const zIndexStyle = computed(() => (props.zIndex ? { zIndex: props.zIndex } : null));
+
+onBeforeUnmount(() => {
+  document.removeEventListener('keyup', escapeIsPressed);
```

Comment:
I'd rename the function `escapeIsPressed` to `onKeyUp`. The function is not executed when the escape is pressed but when the key gets released and it's checking whether the ESC got pressed inside.